README.md
## Get country info by phone number

### Usage

```javascript
import cbp from 'country_by_phone'; // All
import {
  getLocation,
} from 'country_by_phone';          // Separate

console.log(cbp.getIso2('+7705311'));
// KZ

console.log(getLocation('+7705311'));
// Kazakhstan
```

### Remarks

- Full length of the phone number is not required. Five digits are enough
- Possible phone formats (equivalent):
  * _xxxxx_
  * **+**_xxxxx_
  * **00**_xxxxx_
- `require` not supported yet


CHANGELOG.md
#### 0.1.1 (2022-10-26)

##### Documentation Changes

* **usage:**
  *  add remarks ([55b2c667](https://github.com/axsmak/country_by_phone/commit/55b2c667a1363d861dd738bad9df6ba81e763b74))
  *  add basic usage ([434aae66](https://github.com/axsmak/country_by_phone/commit/434aae66ad54d807d0d3c1ff78448781e47c518d))

##### Other Changes (feature)

* **phone:**  plus not required ([9dc3e7cb](https://github.com/axsmak/country_by_phone/commit/9dc3e7cbf8e8531658d61d71d04703bd8f657eeb))


test/index.js
import {
  getLocalInfo,
} from './../index.js';
import phones from './phones.js';

export const list = Object.keys(phones).sort((a, b) => b.length - a.length);

for (let code of list) {
  if (phones.hasOwnProperty(code)) {
    let phone = code;
    for (let i = phone.length; i < 5; i++) {
      phone += '3';
    }
    const countries = getCountries(phone);
    const info = getLocalInfo(`+${phone}`);
    if (info.valid) {
      if (countries.indexOf(info.country_info.iso2) < 0) {
        if (info.country_info.iso2 === 'US') {
          console.info(`INFO TEST: USA area code +${code} expected in`, countries);
        } else {
          console.error(`WARN TEST: Wrong country ${info.country_info.iso2} for code +${code} expected in`, countries);
        }
      }
    } else {
      console.error(`FAIL TEST: Invalid code +${code} expected`, countries, info);
    }
  }
}

function getCountries(phone) {
  let starts = String(phone);
  let codeLength = 0;

  for (let code of list) {
    if (code.length !== codeLength) {
      codeLength = code.length;
      starts = starts.substring(0, codeLength);
    }

    if (code === starts) {
      return phones[code];
    }
  }
  return null;
}


test/phones.js
export default {
  '1242': ['BS'],
  '1246': ['BB'],
  '1264': ['AI'],
  '1268': ['AG'],
  '1284': ['VG'],
  '1340': ['VI'],
  '1345': ['KY'],
  '1441': ['BM'],
  '1473': ['GD'],
  '1649': ['TC'],
  '1658': ['JM'],
  '1664': ['MS'],
  '1670': ['MP'],
  '1671': ['GU'],
  '1684': ['AS'],
  '1721': ['SX'],
  '1758': ['LC'],
  '1767': ['DM'],
  '1784': ['VC'],
  '1787': ['PR'],
  '1809': ['DO'],
  '1829': ['DO'],
  '1849': ['DO'],
  '1868': ['TT'],
  '1869': ['KN'],
  '1876': ['JM'],
  '1939': ['PR'],
  '20': ['EG'],
  '27': ['ZA'],
  '211': ['SS'],
  '212': ['MA', 'EH'],
  '213': ['DZ'],
  '216': ['TN'],
  '218': ['LY'],
  '220': ['GM'],
  '221': ['SN'],
  '222': ['MR'],
  '223': ['ML'],
  '224': ['GN'],
  '225': ['CI'],
  '226': ['BF'],
  '227': ['NE'],
  '228': ['TG'],
  '229': ['BJ'],
  '230': ['MU'],
  '231': ['LR'],
  '232': ['SL'],
  '233': ['GH'],
  '234': ['NG'],
  '235': ['TD'],
  '236': ['CF'],
  '237': ['CM'],
  '238': ['CV'],
  '239': ['ST'],
  '240': ['GQ'],
  '241': ['GA'],
  '242': ['CG'],
  '243': ['CD'],
  '244': ['AO'],
  '245': ['GW'],
  '246': ['IO'],
  '247': ['AC'],
  '248': ['SC'],
  '249': ['SD'],
  '250': ['RW'],
  '251': ['ET'],
  '252': ['SO'],
  '253': ['DJ'],
  '254': ['KE'],
  '255': ['TZ'],
  '256': ['UG'],
  '257': ['BI'],
  '258': ['MZ'],
  '260': ['ZM'],
  '261': ['MG'],
  '262': ['RE', 'YT', 'TF'],
  '263': ['ZW'],
  '264': ['NA'],
  '265': ['MW'],
  '266': ['LS'],
  '267': ['BW'],
  '268': ['SZ'],
  '269': ['KM'],
  '290': ['SH', 'TA'],
  '291': ['ER'],
  '297': ['AW'],
  '298': ['FO'],
  '299': ['GL'],
  '30': ['GR'],
  '31': ['NL'],
  '32': ['BE'],
  '33': ['FR'],
  '34': ['ES'],
  '36': ['HU'],
  '39': ['IT', 'VA'],
  '350': ['GI'],
  '351': ['PT'],
  '352': ['LU'],
  '353': ['IE'],
  '354': ['IS'],
  '355': ['AL'],
  '356': ['MT'],
  '357': ['CY'],
  '358': ['FI', 'AX'],
  '359': ['BG'],
  '370': ['LT'],
  '371': ['LV'],
  '372': ['EE'],
  '373': ['MD'],
  '374': ['AM', 'QN'],
  '375': ['BY'],
  '376': ['AD'],
  '377': ['MC'],
  '378': ['SM'],
  '379': ['VA'],
  '380': ['UA'],
  '381': ['RS'],
  '382': ['ME'],
  '383': ['XK'],
  '385': ['HR'],
  '386': ['SI'],
  '387': ['BA'],
  '389': ['MK'],
  '40': ['RO'],
  '41': ['CH'],
  '43': ['AT'],
  '44': ['GB', 'GG'],
  '45': ['DK'],
  '46': ['SE'],
  '47': ['NO', 'SJ', 'BV'],
  '48': ['PL'],
  '49': ['DE'],
  '420': ['CZ'],
  '421': ['SK'],
  '423': ['LI'],
  '51': ['PE'],
  '52': ['MX'],
  '53': ['CU'],
  '54': ['AR'],
  '55': ['BR'],
  '56': ['CL'],
  '57': ['CO'],
  '58': ['VE'],
  '500': ['FK', 'GS'],
  '501': ['BZ'],
  '502': ['GT'],
  '503': ['SV'],
  '504': ['HN'],
  '505': ['NI'],
  '506': ['CR'],
  '507': ['PA'],
  '508': ['PM'],
  '509': ['HT'],
  '590': ['GP', 'BL', 'MF'],
  '591': ['BO'],
  '592': ['GY'],
  '593': ['EC'],
  '594': ['GF'],
  '595': ['PY'],
  '596': ['MQ'],
  '597': ['SR'],
  '598': ['UY'],
  '599': ['BQ', 'CW'],
  '60': ['MY'],
  '61': ['AU', 'CX', 'CC'],
  '62': ['ID'],
  '63': ['PH'],
  '64': ['NZ', 'PN'],
  '65': ['SG'],
  '66': ['TH'],
  '670': ['TL'],
  '672': ['NF', 'AQ', 'HM'],
  '673': ['BN'],
  '674': ['NR'],
  '675': ['PG'],
  '676': ['TO'],
  '677': ['SB'],
  '678': ['VU'],
  '679': ['FJ'],
  '680': ['PW'],
  '681': ['WF'],
  '682': ['CK'],
  '683': ['NU'],
  '685': ['WS'],
  '686': ['KI'],
  '687': ['NC'],
  '688': ['TV'],
  '689': ['PF'],
  '690': ['TK'],
  '691': ['FM'],
  '692': ['MH'],
  '7': ['RU', 'KZ'],
  '73': ['RU'],
  '74': ['RU'],
  '76': ['KZ'],
  '77': ['KZ'],
  '78': ['RU'],
  '79': ['RU'],
  '81': ['JP'],
  '82': ['KR'],
  '84': ['VN'],
  '86': ['CN'],
  '800': ['XT'],
  '808': ['XS'],
  '850': ['KP'],
  '852': ['HK'],
  '853': ['MO'],
  '855': ['KH'],
  '856': ['LA'],
  '870': ['XN'],
  '878': ['XP'],
  '880': ['BD'],
  '881': ['XG'],
  '882': ['XV'],
  '883': ['XV'],
  '886': ['TW'],
  '90': ['TR', 'CT'],
  '91': ['IN'],
  '92': ['PK'],
  '93': ['AF'],
  '94': ['LK'],
  '95': ['MM'],
  '98': ['IR'],
  '960': ['MV'],
  '961': ['LB'],
  '962': ['JO'],
  '963': ['SY'],
  '964': ['IQ'],
  '965': ['KW'],
  '966': ['SA'],
  '967': ['YE'],
  '968': ['OM'],
  '970': ['PS'],
  '971': ['AE'],
  '972': ['IL'],
  '973': ['BH'],
  '974': ['QA'],
  '975': ['BT'],
  '976': ['MN'],
  '977': ['NP'],
  '979': ['XR'],
  '991': ['XC'],
  '992': ['TJ'],
  '993': ['TM'],
  '994': ['AZ'],
  '995': ['GE'],
  '996': ['KG'],
  '997': ['KZ'],
  '998': ['UZ'],
};


index.js
import canada from './lib/canada.js';
import countries from './lib/countries.js';
import us from './lib/us.js';

// Add location
function addLocation(localInfo) {

  let text = localInfo.text;
  let location = false;

  switch (localInfo.type) {
    case 'area':
      location = getLoc(localInfo);
      while (!location && text.length > 3) {
        text = text.substring(0, text.length - 1);
        localInfo.text = text;
        location = getLoc(localInfo);
      }
      if (location) {
        chooseUsCanada(localInfo);
      }
      break;

    case 'country':
      localInfo.country_info = getCountryInfo(localInfo);
      location = localInfo.country_info.name;
      while (!location && text.length > 2) {
        text = text.substring(0, text.length - 1);
        localInfo.text = text;
        localInfo.country_info = getCountryInfo(localInfo);
        location = localInfo.country_info.name;
      }
      break;
  }

  if (!location) {
    localInfo.valid = false;
  }

  localInfo.location = location;
}

//Detect if Area Code or Country Code
function addType(localInfo) {
  const text = localInfo.text;
  const beg = text.substring(0, 1);
  const beg2 = text.substring(0, 2);
  let type = '';
  if (beg === '+' || beg2 === '00') {
    localInfo.text = text.substring(0, 5);
    type = 'country';
  } else {
    if (text.length < 3) {
      localInfo.valid = false;
    }
    localInfo.text = text.substring(0, 3);
    type = 'area';
  }
  localInfo.type = type;
}

//Check for US to process as area
function checkUS(localInfo) {
  let text = localInfo.text;
  const beginning = text.substring(0, 2);

  if (text !== '+1') {
    if (beginning === '+1') {
      text = text.replace('+1', '');
    }
    if (text.substring(0, 1) == '1') {
      text = text.substring(1, text.length);
    }
    if (text.length < 3) {
      localInfo.valid = false;
    }
  }

  localInfo.text = text;
}

//For US/Canada Area codes, add correct Country
function chooseUsCanada(localInfo) {

  const isCanada = canada.indexOf(localInfo.text) >= 0;
  const usIndex = getCountryInfo({
    text: '+1',
  })['index'];
  localInfo.country_info = countries[usIndex + (isCanada ? 2 : 1)];
}

//Allow only numerical and '+' characters
function cleanText(x) {
  let text = x.replace(/[^+0-9]/g, '');
  if (text.length > 5) {
    text = text.substring(0, 5);
  }
  return text;
}

// Get country info
function getCountryInfo(localInfo) {
  const code = localInfo.text.slice(1);
  const countryIndex = countries.findIndex((country) => country.code === code);

  if (countryIndex >= 0) {
    const countryInfo = countries[countryIndex];

    Object.defineProperty(countryInfo, 'index', {
      enumerable: true,
      value: countryIndex,
    });

    return countryInfo;
  }

  return {
    index: null,
  };
}

//Linear Search US Time Zones
function getLoc(localInfo) {
  const text = localInfo.text;
  let location = null;

  for (let usArea of us) {
    for (let x = 0; x < usArea.codes.length; x++) {
      if (text === usArea.codes[x]) {
        location = usArea.codes[x + 1];
        break;
      }
    }
  }

  if (location) {
    return location;
  }

  const tollFree = us[us.length - 1];
  for (let code of tollFree.codes) {
    if (code === text) {
      location = 'Toll Free or Other';
      break;
    }
  }

  return location;
}

export function getIso2(phone) {
  const localInfo = getLocalInfo(phone);
  return localInfo.valid ? localInfo.country_info.iso2 : null;
}

export function getLocation(phone) {
  const localInfo = getLocalInfo(phone);
  return localInfo.valid ? localInfo.location : null;
}

//Build Local Info object
export function getLocalInfo(phone, callback) {
  if (!phone) {
    return;
  }
  let text = standardizeCountry(cleanText(String(phone)));
  const localInfo = {
    text,
    valid: true,
    country_info: {
      index: null,
      iso2: null,
    },
    location: null,
  };
  //Check for US
  checkUS(localInfo);
  //Add Type (Area or Country)
  addType(localInfo);
  //Add Location and Country Info
  addLocation(localInfo);

  if (typeof callback === 'function') {
    callback(localInfo);
    return;
  }
  return localInfo;
}

// Correct 00 and replace for +
function standardizeCountry(x) {
  if (x.substring(0, 2) === '00') {
    x = x.replace('00', '+');
  }
  return x[0] === '+' ? x : `+${x}`;
}

export default {
  getIso2,
  getLocalInfo,
  getLocation,
};


lib/us.js
export default [
  {
    name: 'AST',
    dst: true,
    offset: (-4),
    codes: ['441', 'Bermuda', '506', 'New Brunswick', '782', 'Nova Scotia', '902', 'Nova Scotia']
  },
  {
    name: 'AST',
    dst: false,
    offset: (-4),
    codes: ['758', 'Saint Lucia', '246', 'Barbados', '264', 'Anguilla', '268', 'Antigua and Barbuda', '284', 'British Virgin Islands', '340', 'U.S. Virgin Islands', '473', 'Grenada', '664', 'Montserrat', '721', 'Sint Maarten', '767', 'Dominica', '784', 'Saint Vincent & the Grenadines', '787', 'Puerto Rico', '868', 'Trinidad and Tobago', '869', 'Saint Kitts & Nevis', '939', 'Puerto Rico']
  },
  {
    name: 'NST',
    dst: true,
    offset: (-3.5),
    codes: ['709', 'Newfoundland']
  },
  {
    name: 'EST',
    dst: true,
    offset: (-5),
    codes: ['463', 'Indianapolis, IN', '934', 'Suffolk, NY', '548', 'Toronto, ON', '743', 'Greensboro, NC', '854', 'Charleston, SC', '380', 'Columbus, OH', '220', 'Columbus, OH', '930', 'Evansville, IN', '201', 'Newark, NJ', '202', 'Washington, DC', '203', 'New Haven, CT', '207', 'Portland, ME', '212', 'New York, NY', '215', 'Philadelphia, PA', '216', 'Cleveland, OH', '223', 'Lancaster, PA', '226', 'Elora, ON', '229', 'Albany, GA', '231', 'Muskegon, MI', '234', 'Akron, OH', '239', 'Fort Meyers, FL', '240', 'Baltimore, MD', '242', 'Bahamas', '248', 'Detroit, MI', '249', 'Algoma Mills, ON', '252', 'Greenville, NC', '260', 'Fort Wayne, IN', '263', 'Montreal, QC', '267', 'Philadelphia, PA', '269', 'Kalamazoo, MI', '272', 'Lake Ariel, PA', '276', 'Martinsville, VA', '283', 'Cincinnati, OH', '289', 'Mount Albert, ON', '301', 'Baltimore, MD', '302', 'Wilmington, DE', '304', 'Charleston, WV', '305', 'Miami, FL', '313', 'Detroit, MI', '315', 'Syracuse, NY', '317', 'Indianapolis, IN', '321', 'Orlando, FL', '326', 'Dayton, OH', '329', 'Poughkeepsie, NY', '330', 'Akron, OH', '332', 'New York, NY', '336', 'Greensboro, NC', '339', 'Boston, MA', '343', 'Northbrook, ON', '347', 'Brooklyn, NY', '351', 'Danvers, MA', '352', 'Gainesville, FL', '363', 'Hempstead, NY', '354', 'Granby, QC', '365', 'Bradford, ON', '367', 'Quebec City, QC', '382', 'London, ON', '386', 'Daytona Beach, FL', '401', 'Providence, RI', '404', 'Atlanta, GA', '407', 'Orlando, FL', '410', 'Baltimore, MD', '412', 'Pittsburgh, PA', '413', 'Springfield, MA', '416', 'Toronto, ON', '418', 'Chambord, QC', '419', 'Toledo, OH', '434', 'Lynchburg, VA', '437', 'Toronto, ON', '438', 'Montreal, QC', '440', 'Cleveland, OH', '443', 'Baltimore, MD', '445', 'Philadelphia, PA', '450', 'Saint Remi, QC', '468', 'Sherbrooke, QC', '470', 'Atlanta, GA', '472', 'Fayetteville, NC', '475', 'New Haven, CT', '478', 'Macon, GA', '428', 'Moncton, NB', '484', 'Philadelphia, PA', '502', 'Louisville, KY', '508', 'Boston, MA', '513', 'Cincinnati, OH', '514', 'Montreal, QC', '516', 'Springfield Gardens, NY', '517', 'Lansing, MI', '518', 'Schenectady, NY', '519', 'Harriston, ON', '540', 'Fredericksburg, VA', '551', 'Newark, NJ', '561', 'West Palm Beach, FL', '567', 'Toledo, OH', '570', 'Wilkes Barre, PA', '571', 'Alexandria, VA', '579', 'Saint Rosalie, QC', '581', 'Baie Johan Beetz, QC', '582', 'Erie, PA', '585', 'Buffalo, NY', '586', 'Detroit, MI', '603', 'Manchester, NH', '607', 'Freeville, NY', '609', 'Camden, NJ', '610', 'Philadelphia, PA', '613', 'Clarence Creek, ON', '614', 'Columbus, OH', '616', 'Grand Rapids, MI', '617', 'Boston, MA', '631', 'Deer Park, NY', '640', 'Trenton, NJ', '645', 'Miami, FL', '646', 'Brooklyn, NY', '647', 'Toronto, ON', '656', 'Tampa, FL', '658', 'Jamaica', '649', 'Turks and Caicos Islands', '667', 'Baltimore, MD', '678', 'Atlanta, GA', '680', 'Syracuse, NY', '681', 'Charleston, WV', '683', 'Sudbury, ON', '689', 'Orlando, FL', '703', 'Alexandria, VA', '704', 'Charlotte, NC', '705', 'Kapuskasing, ON', '706', 'Augusta, GA', '716', 'Buffalo, NY', '717', 'Lancaster, PA', '718', 'Brooklyn, NY', '724', 'Pittsburgh, PA', '727', 'Tampa, FL', '728', 'West Palm Beach, FL', '732', 'Newark, NJ', '734', 'Detroit, MI', '740', 'Columbus, OH', '742', 'Hamilton, ON', '753', 'Northbrook, ON', '754', 'Fort Lauderdale, FL', '757', 'Virginia Beach, VA', '762', 'Augusta, GA', '765', 'Indianapolis, IN', '770', 'Atlanta, GA', '771', 'Washington, DC', '772', 'Port Saint Lucie, FL', '774', 'Boston, MA', '781', 'Boston, MA', '786', 'Miami, FL', '802', 'Shelburne, VT', '803', 'Columbia, SC', '804', 'Richmond, VA', '809', 'Dominican Republic', '810', 'Flint, MI', '813', 'Tampa, FL', '814', 'Erie, PA', '819', 'Gatineau, QC', '826', 'Roanoke, VA', '828', 'Asheville, NC', '829', 'Dominican Republic', '835', 'Allentown, PA', '838', 'Albany, NY', '839', 'Columbia, SC', '843', 'Charleston, SC', '845', 'Poughkeepsie, NY', '848', 'Newark, NJ', '849', 'Dominican Republic', '856', 'Camden, NJ', '857', 'Boston, MA', '859', 'Lexington, KY', '860', 'Hartford, CT', '862', 'Newark, NJ', '863', 'Kissimmee, FL', '864', 'Greenville, SC', '865', 'Knoxville, TN', '873', 'Lannoncatn, QC', '876', 'Jamaica', '878', 'Pittsburgh, PA', '904', 'Jacksonville, FL', '905', 'Palgrave, ON', '908', 'Newark, NJ', '910', 'Fayetteville, NC', '912', 'Savannah, GA', '914', 'Manhattan, NY', '917', 'Brooklyn, NY', '919', 'Raleigh, NC', '929', 'Brooklyn, NY', '937', 'Dayton, OH', '941', 'Sarasota, FL', '943', 'Atlanta, GA', '947', 'Southfield, MI', '948', 'Virginia Beach, VA', '954', 'Fort Lauderdale, FL', '959', 'Hartford, CT', '973', 'Newark, NJ', '978', 'Boston, MA', '980', 'Charlotte, NC', '984', 'Raleigh, NC', '989', 'Saginaw, MI']
  },
  {
    name: 'EST',
    dst: false,
    offset: (-5),
    codes: ['345', 'Cayman Islands']
  },
  { //Arizona Times
    name: 'MST',
    dst: false,
    offset: (-7),
    codes: ['602', 'Phoenix, AZ', '520', 'Phoenix, AZ', '480', 'Phoenix, AZ', '623', 'Phoenix, AZ', '928', 'Phoenix, AZ']
  },
  {
    name: 'CST',
    dst: true,
    offset: (-6),
    codes: ['726', 'San Antonio, TX', '737', 'Austin, TX', '204', 'Strathclair, MB', '205', 'Birmingham, AL', '210', 'San Antonio, TX', '214', 'Dallas, TX', '217', 'Chicago, IL', '218', 'Duluth, MN', '224', 'Chicago, IL', '225', 'Baton Rouge, LA', '228', 'Gulfport, MS', '251', 'Mobile, AL', '254', 'Killeen, TX', '256', 'Hunstville, AL', '262', 'Milwaukee, WI', '274', 'Green Bay, WI', '281', 'Houston, TX', '306', 'Gravelborg, SK', '309', 'Peoria, IL', '312', 'Chicago, IL', '314', 'Saint Louis, MO', '316', 'Wichita, KS', '318', 'Shreveport, LA', '319', 'Cedar Rapids, IA', '320', 'Saint Cloud, MN', '325', 'Abilene, TX', '327', 'Jonesboro, AR', '331', 'Lemont, IL', '334', 'Montgomery, AL', '337', 'Lafayette, LA', '346', 'Houston, TX', '361', 'Corpus Christi, TX', '402', 'Omaha, NE', '405', 'Oklahoma City, OK', '409', 'Beaumont, TX', '414', 'Milwaukee, WI', '417', 'Springfield, MO', '430', 'Longview, TX', '431', 'Killarney, MB', '432', 'Midland, TX', '447', 'Springfield, IL', '464', 'Chicago, IL', '469', 'Dallas, TX', '474', 'Saskatoon, SK', '479', 'Springdale, AR', '501', 'Little Rock, AR', '504', 'New Orleans, LA', '507', 'Rochester, MN', '512', 'Austin, TX', '515', 'Des Moines, IA', '531', 'Lincoln, NE', '534', 'Chippewa Falls, WI', '539', 'Tulsa, OK', '557', 'Saint Louis, MO', '563', 'Davenport, IA', '572', 'Oklahoma City, OK', '573', 'Columbia, MO', '580', 'Oklahoma City, OK', '584', 'Winnipeg, MB', '601', 'Jackson, MS', '608', 'Madison, WI', '612', 'Minneapolis, MN', '615', 'Nashville, TN', '629', 'Nashville, TN', '618', 'Belleville, IL', '625', 'Nashville, TN', '630', 'Chicago, IL', '636', 'Saint Louis, MO', '639', 'Saskatoon, SK', '641', 'Beaman, IA', '651', 'Minneapolis, MN', '659', 'Birmingham, AL', '660', 'Otterville, MO', '662', 'Southaven, MS', '682', 'Dallas, TX', '708', 'Chicago, IL', '712', 'Sioux City, IA', '713', 'Houston, TX', '715', 'Green Bay, WI', '726', 'San Antonio, TX', '731', 'Jackson, TN', '763', 'Minneapolis, MN', '769', 'Jackson, MS', '773', 'Chicago, IL', '779', 'Rockford, IL', '806', 'Lubbock, TX', '815', 'Chicago, IL', '816', 'Kansas City, MO', '817', 'Dallas, TX', '830', 'New Braunfels, TX', '832', 'Houston, TX', '847', 'Chicago, IL', '861', 'Peoria, IL', '870', 'Jonesboro, AR', '872', 'Chicago, IL', '901', 'Memphis, TN', '903', 'Tyler, TX', '913', 'Kansas City, KS', '918', 'Tulsa, OK', '920', 'Milwaukee, WI', '936', 'Conroe, TX', '938', 'Goodwater, AL', '940', 'Denton, TX', '945', 'Dallas, TX', '952', 'Minneapolis, MN', '956', 'Laredo, TX', '972', 'Dallas, TX', '975', 'Kansas City, MO', '979', 'College Station, TX', '985', 'New Orleans, LA']
  },
  {
    name: 'MST',
    dst: true,
    offset: (-7),
    codes: ['825', 'Edmonton, Alberta', '303', 'Denver, CO', '307', 'Casper, WY', '368', 'Edmonton, Alberta', '385', 'Salt Lake City, UT', '403', 'Gadsby, Alberta', '406', 'Billings, MT', '435', 'Park City, UT', '505', 'Albuquerque, NM', '575', 'Las Cruces, NM', '587', 'Fahler, Alberta', '719', 'Colorado Springs, CO', '720', 'Denver, CO', '780', 'Beaver Lodge, Alberta', '801', 'Salt Lake City, UT', '970', 'Fort Collins, CO', '983', 'Denver, CO']
  },
  {
    name: 'MST/CST',
    dst: true,
    offset: [(-6), (-7)],
    codes: ['915', 'El Paso, TX', '308', 'Grand Island, NE', '605', 'Sioux Falls, SD', '620', 'Hutchinson, KS', '701', 'Fargo, ND', '785', 'Topeka, KS', '867', 'Pont Inlet, NT']
  },
  {
    name: 'CST/EST',
    dst: true,
    offset: [(-5), (-6)],
    codes: ['219', 'Valparaiso, IN', '270', 'Bowling Green, KY', '364', 'Bowling Green, KY', '448', 'Pensacola, FL', '807', 'Manitouwadge, ON', '812', 'Evansville, IN', '850', 'Pensacola, FL', '931', 'Clarksville, TN', '423', 'Chattanooga, TN', '574', 'South Bend, IN', '606', 'Sharpsburg, KY', '906', 'Deerton, MI']
  },
  {
    name: 'PST/MST',
    dst: true,
    offset: [(-7), (-8)],
    codes: ['208', 'Boise, ID', '458', 'Eugene, OR', '541', 'Eugene, OR', '986', 'Boise, ID']
  },
  {
    name: 'PST',
    dst: true,
    offset: (-8),
    codes: ['778', 'Vancouver, BC', '628', 'San Francisco, CA', '206', 'Seattle, WA', '209', 'Fresno, CA', '213', 'Los Angeles, CA', '236', 'Victoria, BC', '250', 'Mcleese Lake, BC', '253', 'Seattle, WA', '279', 'Sacramento, CA', '310', 'Los Angeles, CA', '323', 'Los Angeles, CA', '341', 'Oakland, CA', '350', 'Fresno, CA', '360', 'Seattle, WA', '408', 'San Jose, CA', '415', 'San Francisco, CA', '424', 'Los Angeles, CA', '425', 'Seattle, WA', '442', 'Victorville, CA', '503', 'Portland, OR', '509', 'Spokane, WA', '510', 'San Jose, CA', '530', 'Sacramento, CA', '559', 'Fresno, CA', '562', 'Los Angeles, CA', '564', 'Seattle, WA', '604', 'Mission, BC', '619', 'San Diego, CA', '626', 'Los Angeles, CA', '650', 'San Jose, CA', '657', 'Long Beach, CA', '661', 'Bakersfield, CA', '669', 'San Jose, CA', '672', 'Vancouver, BC', '702', 'Las Vegas, NV', '707', 'Santa Rosa, CA', '714', 'Long Beach, CA', '725', 'Las Vegas, NV', '747', 'Los Angeles, CA', '760', 'San Diego, CA', '775', 'Reno, NV', '778', 'Wellington, BC', '805', 'Bakersfield, CA', '818', 'Los Angeles, CA', '820', 'Bakersfield, CA', '831', 'Salinas, CA', '840', 'San Bernardino, CA', '858', 'San Diego, CA', '909', 'Riverside, CA', '916', 'Sacramento, CA', '925', 'Oakland, CA', '949', 'Santa Ana, CA', '951', 'Riverside, CA', '971', 'Portland, OR']
  },
  {
    name: 'SST',
    dst: false,
    offset: (-11),
    codes: ['684', 'American Samoa']
  },
  {
    name: 'HAST',
    dst: false,
    offset: (-10),
    codes: ['808', 'Hawaii']
  },
  {
    name: 'Chamorro Standard Time',
    dst: false,
    offset: (10),
    codes: ['670', 'Common Wealth of the Northern Mariana Islands']
  },
  {
    name: 'GUAM',
    dst: false,
    offset: (10),
    codes: ['671', 'Guam']
  },
  {
    name: 'AKST',
    dst: true,
    offset: (-9),
    codes: ['907', 'Juneau']
  },
  { //Keep as Last Index
    name: 'Toll Free or Other',
    dst: false,
    offset: false,
    codes: ['911', '456', '500', '521', '523', '524', '525', '526', '527', '528', '533', '544', '566', '577', '600', '700', '710', '800', '844', '855', '866', '877', '833', '888', '900']
  },
];
